assets/js/state/lastExecutions.js

```diff
@@ -0,0 +1,62 @@
+import { createSlice } from '@reduxjs/toolkit';
+
+const initialState = {};
+
+const initialExecutionState = {
+  data: null,
+  loading: false,
+  error: null,
+};
+
+export const lastExecutionsSlice = createSlice({
+  name: 'lastExecutions',
+  initialState,
+  reducers: {
+    setLastExecutionLoading: (state, { payload: groupID }) => {
+      const lastExecutionState = {
+        ...initialExecutionState,
+        loading: true,
+      };
+
+      state[groupID] = lastExecutionState;
+    },
+    setLastExecutionEmpty: (state, { payload: groupID }) => {
+      const lastExecutionState = {
+        ...initialExecutionState,
+      };
+
+      state[groupID] = lastExecutionState;
+    },
+    setLastExecution: (state, { payload }) => {
+      const { group_id: groupID } = payload;
+
+      const lastExecutionState = {
+        ...initialExecutionState,
+        data: payload,
+        loading: false,
```

Comment:
You can omit this `loading` here, the same in `setLastExecutionError`. They are false by default